tests/test_rtmbot_core.py

```diff
@@ -18,3 +28,32 @@ def test_init():
     l.check(
         ('root', 'INFO', 'Initialized in: /tmp/')
     )
+
+def test_output():
+    ''' Test that sending a message behaves as expected '''
+    rtmbot = init_rtmbot()
+
+    # Mock the slack_client object
+    slackclient_mock = Mock()
```

Comment:
You want to [`create_autospec`](https://github.com/testing-cabal/mock/blob/master/mock/mock.py#L2262) here.

```
from slackclient import SlackClient

slackclient_mock = mock.create_autospec(SlackClient, spec_set=True)
```

For the above and all other mocks. This is going to save you from making a typo somewhere and actually ensure that you're writing tests against real methods. https://docs.python.org/3/library/unittest.mock.html#auto-speccing has a good write up on more details- happy to discuss further, since it took me a long while to grok!